Bindings/Java/Matlab/Hopper_Device/ModifyInteractiveHopperSolution.m

```diff
@@ -0,0 +1,166 @@
+
+import org.opensim.modeling.*;
+
+%% HOPPER AND DEVICE SETTINGS
+
+additionalMass = 0;
+
+% Retrieve muscle settings based on user selection 
+%   default: "The Average Joe"
+[muscleFunc] = InteractiveHopperSettings(muscle);
+[maxIsometricForce,tendonStiffness,tendonSlackLength,muscleMass] = muscleFunc();
+MillardTendonParams = [0.049 tendonStiffness 0.67 0.5 tendonSlackLength];
+additionalMass = additionalMass + muscleMass;
+
+% Retreive passive device settings if passive device specified
+%   default: no passive device
+%            if device --> passiveParameter = 50
+if addPassiveDevice
+    passive = InteractiveHopperSettings('passive');
+    [passiveMass,springStiffness] = passive(passiveParameter);
+    additionalMass = additionalMass + passiveMass;
+end
+
+% Retreive passive device settings if passive device specified
+%   default: no active device
+%            if device --> activeParameter = 50
+if addActiveDevice
+    if isActivePropMyo
+        activePropMyo = InteractiveHopperSettings('activePropMyo');
+        [activeMass,gain] = activePropMyo(activeParameter);
+    else
+        activeControl = InteractiveHopperSettings('activeControl');
+        [activeMass,maxTension] = activeControl(activeParameter);
+    end
+    additionalMass = additionalMass + activeMass;
+end
+
+%% MODIFY HOPPER MODEL
+
+% Build hopper
+hopper = ModifyHopper(hopper, ...
+                     'excitation',muscleExcitation, ...
+                     'additionalMass',additionalMass, ...
+                     'MillardTendonParams', MillardTendonParams, ...
+                     'maxIsometricForce', maxIsometricForce);
+hopper.printSubcomponentInfo();
+
+%% MODIFY DEVICES
+devices = cell(0);
+deviceNames = cell(0);
+patellaWrap = cell(0);
+
+% Passive device
+if addPassiveDevice
+    passive = ModifyDevice(hopper, ...
+                          'deviceType','passive', ... 
+                          'springStiffness',springStiffness, ...
+                          'passivePatellaWrap',passivePatellaWrap);
+    devices{1,length(devices)+1} = passive;
+    deviceNames{1,length(deviceNames)+1} = 'device_passive';
+    patellaWrap{1,length(patellaWrap)+1} = passivePatellaWrap;  
+end
+
+% Active device
+if addActiveDevice
+    if isActivePropMyo
+        active = ModifyDevice(hopper, ...
+                             'deviceType','active', ...
+                             'isPropMyo',true, ...
+                             'gain',gain);
+    else
+        active = ModifyDevice(hopper, ...
+                             'deviceType','active', ...
+                             'isPropMyo',false, ... 
+                             'control',deviceControl, ...
+                             'maxTension',maxTension);
+    end
+    
+    devices{1,length(devices)+1} = active;
+    deviceNames{1,length(deviceNames)+1} = 'device_active';
+    patellaWrap{1,length(patellaWrap)+1} = activePatellaWrap;   
+end
+
+end
+
+function ModifyHopper(hopper, varargin)
```

Comment:
This isn't currently being used. I don't plan on fleshing this out unless it's helpful for speed up @carmichaelong's optimization example. Otherwise, I'm fine with removing it.